File: ReleaseNotesGenerator/Domain/Branch.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ReleaseNotesGenerator.Domain
{
    public class Branch : EntityBase<int>
    {
        [Required]
        public string Name { get; set; }

        public string LastCommitId { get; set; }

        public DateTime? LastCommitDateTime { get; set; }

        [Required]
        public int RepositoryId { get; set; }

        public Repository Repository { get; set; }

        public List<RepositoryItemPath> RepositoryItemPaths { get; set; }
    }
}

File: ReleaseNotesGenerator/ClientApp/app/common/classes/Branch.ts

    import { Repository } from './Repository';
import { RepositoryItemPath } from './RepositoryItemPath';
import { EntityBase } from './EntityBase';
    export class Branch extends EntityBase<number> {
        
        name: string;
        lastCommitId: string;
        lastCommitDateTime: Date;
        repositoryId: number;
        repository: Repository;
        repositoryItemPaths: RepositoryItemPath[];
        }

    